### DISCUSSION TITLE
Support for Just Using Remote Photos

### DISCUSSION THREAD

**Discussion by ilkersigirci** [immich-app/immich#32] [Q&A] [answered]
Firstly, thank you for this awesome app.  My question may be related to #7, but I would like to make it more clear here. I want to use this app to access my photos on my remote server. Afaik, this is not directly possible in the app since you designed in a way that photos should be backed up from the phone first. Will that be changed in a way that, in the app I can show remote photo location and they are directly be seen in the app without doing any backup?

**Comment by alextran1502** [comment, marked as answer]
With the current setup I think it is not possible, because the server needs to have the information of the photos/videos in the database to query and to perform order functionalities. I think the best solution I can do for this situation and for other scenarios that have photos/video on a different location is a software for importing or auto importing if you point the software to a directory, ether locally or remotely. So that it would perform a walk through the folders and upload those assets to the Immich server. 

I think that would be a balance in cases that people have thousands of assets already stored somewhere and they don't want to overwhelm the web browser by dragging couple terabytes and drop them on the browser. The trade off is that you will temporarily have two storage locations while the assets are being transferred over. 

However, since this is not the primary goal I set out to do, my focus right now is still a good backup tool for the assets already on your mobile phone :). But yeah, I will keep this in my future plan for the application. 

Thank you for a very good question!

**Reply by JM-Lemmi** [reply]
> So that it would perform a walk through the folders and **upload** those assets to the Immich server.

> The trade off is that you will temporarily have **two storage locations** while the assets are being transferred over.

Do I read your answer correct, that you would plan on having the immich server be the master storage with its own storage location?

If you ever come around to this feature, maybe consider having all the storage be external? This would mean that immich scans the mounted storage for photos and populates its database, but does not copy the files to itself. The master is still on the network share and immich is just the frontend for displaying the pictures on the go.

Right now I would solve this by having the NAS do deduplication, so having the original master files and the immich volume on the same drive and then deduplicating on OS level, but this of course increases the load on the NAS.

Really looking forward to further development.

**Comment by ilkersigirci** [comment]
Thank you for the detailed and the quick response. I just love the app. I just wish it can access the photos on my server. I will definitely looking forward to the future updates about this.

**Comment by ippocratis** [comment]
Any sort of solution to add remote photos would be welcome
the only workaround for now is kinda insane lol (transfer photos from remote to phone so it can be transfered to immich)
Still this is my personal wanted feature so I can migrate to immich

**Comment by ilkersigirci** [comment]
Just a small feature request about this subject:
- Could you implement the backup mechanism to preserve folder name?
- For example, I have `DCIM/Camera` and `Pictures/Telegram` folders in my phone. When I do backup, in my remote server, these photos are located as merged folder with random name like `$UPLOAD_DIR/0268595a-45c0-4d09-af89-70ecb56b3c63/original/...`. But I want them to be same as the folder structure in my phone.
- With this way, I can easily identify my photos of their appropriate folder names. I think this way is a good approach until we get importing from remote server feature.

**Reply by alextran1502** [reply]
Can you help me open an issue? I will take a look from there. Thanks
